Navigation/Profile/LogInViewController.swift

```diff
@@ -0,0 +1,195 @@
+                scrollView.contentInset.bottom = keyboardSize.height
+                /// Когда клавиатура появляется, verticalScrollIndicatorInsets делает отступ по низу.
+                scrollView.verticalScrollIndicatorInsets = UIEdgeInsets(top: 0, left: 0, bottom: keyboardSize.height, right: 0)
+            }
+        }
+        
+    @objc fileprivate func keyboardWillHide(notification: NSNotification) {
+        /// Когда клавиатура скрывается, scroll view делает отступ до 0.
+            scrollView.contentInset.bottom = .zero
+        /// Когда клавиатура скрывается, verticalScrollIndicatorInsets делает отступ до 0.
+            scrollView.verticalScrollIndicatorInsets = .zero
+            
+        }
+    
+    func setUpLogInView() {
+        view.backgroundColor = UIColor.white
+        
+        view.addSubview(scrollView)
+        
+        scrollView.addSubview(containerView)
+        
+        [logoVK,
+         emailPhoneTextField,
+         passwordTextField,
+         logInButton].forEach {
+            containerView.addSubview($0)
+        }
+        
+        [logoVK,
+         emailPhoneTextField,
+         passwordTextField,
+         logInButton,
+         containerView,
+         scrollView].forEach {
+            $0.translatesAutoresizingMaskIntoConstraints = false
+        }
+        
+        NSLayoutConstraint.activate([
+            scrollView.topAnchor.constraint(equalTo: view.topAnchor),
+            scrollView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            scrollView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            scrollView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            scrollView.heightAnchor.constraint(equalTo: view.heightAnchor),
+            
+            containerView.topAnchor.constraint(equalTo: scrollView.topAnchor),
+            containerView.leadingAnchor.constraint(equalTo: scrollView.leadingAnchor),
+            containerView.trailingAnchor.constraint(equalTo: scrollView.trailingAnchor),
+            containerView.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor),
+            /// Ограниечение прокручивания по ширине container view.
+            containerView.widthAnchor.constraint(equalTo: scrollView.widthAnchor),
+            
+            logoVK.topAnchor.constraint(equalTo: containerView.topAnchor, constant: 120),
+            logoVK.centerXAnchor.constraint(equalTo: containerView.centerXAnchor),
+            logoVK.widthAnchor.constraint(equalToConstant: logoVKWidth),
+            logoVK.heightAnchor.constraint(equalToConstant: logoVKWidth),
+            
+            emailPhoneTextField.topAnchor.constraint(equalTo: logoVK.bottomAnchor, constant: 120),
+            emailPhoneTextField.leadingAnchor.constraint(equalTo: containerView.safeAreaLayoutGuide.leadingAnchor, constant: 16),
+            emailPhoneTextField.trailingAnchor.constraint(equalTo: containerView.safeAreaLayoutGuide.trailingAnchor, constant: -16),
```

Comment:
No need to use safeAreaLayoutGuide here and after.
Logically you would want to use on the first container constraints, which is scrollView in this case 
Other containers will use bounds of parent

Reply:
Done